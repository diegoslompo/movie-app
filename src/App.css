/*SHAME
  SHAME
  SHAME
  SHAME
  SHAME    :( :(
  SHAME
  SHAME
  SHAME
  SHAME */


.ReactModal__Overlay.ReactModal__Overlay--after-open {
  background: rgba(74, 42, 140, 0.8)!important;
}
.ReactModal__Content.ReactModal__Content--after-open {
  width: 280px;
  margin: 0 auto;
  position: relative!important;
  border: 0px!important;
  padding: 0px!important;
  overflow: hidden!important;
  background: #725ea057!important;
  padding-bottom: 15px!important;
}

.ReactModal__Overlay.ReactModal__Overlay--after-open {
  display: flex;
  align-items: center;
  justify-content: center;
}


.ds-movie__details img {
  max-width: 100%;
}

.ds-movie__details p {
  margin: 2px 15px;
  color: #fff;
}

.ds-movie__details h2 {
  color: #fff;
  margin: 5px 15px;
}

.ds-movie__details {
  text-align: center;
}

.ReactModal__Content.ReactModal__Content--after-open button {
  border: 0;
  border-radius: 20px 0 0 0;
  padding: 14px 19px;
  position: absolute;
  bottom: 0px;
  right: 0;
  background: #ff6666;
  color: #fff;
  font-weight: bold;
}
