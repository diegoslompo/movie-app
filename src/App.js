import React, { Component } from 'react'
import * as MoviesAPI from './utils/MoviesAPI'
import styled from 'styled-components'
import './App.css';
// mainly components
import Movie from './components/Movie'
import Notification from './components/Notification'


class App extends Component {

  // {Set} initial state movies
  state = {
    movies: [],
    query:'',
    searchMovies: '',
    searchEmpty: false,
    infoFeat: {
      title: 'Search by',
      description: 'Batman - Avengers - Rick and Morty - [etc...]'
    },
    infoEmpty: {
      title: ':(',
      description: 'SORRY - NO DATABASE API RETURN MOVIES'
    }
  }

  // [getAll] movies the server API use the async + await
  async componentDidMount() {
    const movies = await MoviesAPI.getAll()
    this.setState({ movies })
   }  

  updateQuery = (query) => {
    this.setState({ query: query })
    this.changeQuery(query)
  }

  changeQuery = (resQuery) => {
    if (resQuery !== '') {
      MoviesAPI.search(resQuery).then((result) => {
        if (this.state.query !== '' && result.length > 0) {
          this.setState({ searchMovies: result, searchEmpty: false })
        } else {
          this.setState({ searchMovies: '', searchEmpty: true })
        }
      })
    } else {
      this.setState({ movies: this.state.movies, searchEmpty: false })
    }
  }

  
  render () {

    const { movies, query, searchEmpty, searchMovies, infoFeat, infoEmpty } = this.state 
    const currentMovies = searchMovies !== '' ? searchMovies : movies

    return (
      <Main className="ds-app">
        <Container>
          <Input 
            type="text"
            placeholder="Search"
            value={query}
            onChange={(event) => this.updateQuery(event.target.value)}
          />
          { !searchEmpty ? (
            <div>
              <Notification
                title={infoFeat.title}
                description={infoFeat.description}
              />
              <Movie
                movieItem={currentMovies} 
              />
            </div>
          ):(
            <Notification
              title={infoEmpty.title}
              description={infoEmpty.description}
            />
          )}
        </Container>
      </Main>
    )
  }
} 

const Main = styled.div`
  height: 100%;
  font-family: monospace;
`;

const Container = styled.div`
  margin: 0 auto;
  padding: 20px 0;
  width: 600px;
`;

const Input = styled.input`
  border: 0;
  width: -webkit-fill-available;
  height: 40px;
  padding: 5px 15px;
  margin-top: 20px;
  border-radius: 3px;
  overflow: hidden;
  font-size: 18px;
`;

export default App;
