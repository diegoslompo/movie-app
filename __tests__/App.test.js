import React from "react"
// vai simular a dom em testes, renderizando componentes usando shallow
import { shallow } from "enzyme"
import App from '../src/App'

// descreve o teste no componente
describe('Testing component App', () => {
    // passa a ação desejada de retorno
    it('should render corretly', () => {

        const wrapper = shallow(
            <App />
        )

        expect(wrapper).toMatchSnapshot()
    })
})