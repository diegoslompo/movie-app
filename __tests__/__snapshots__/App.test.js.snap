// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component App should render corretly 1`] = `
<ForwardRef(styled.div)
  className="ds-app"
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.input)
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <div>
      <Notification
        description="Batman - Avengers - Rick and Morty - [etc...]"
        title="Search by"
      />
      <Movie
        movieItem={Array []}
      />
    </div>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;
